File: c/gpt/p124_nre.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        fprintf(stderr, "Error: expected exactly 1 argument\n");
        return 1;
    }

    const char *input = argv[1];
    printf("%s\n", input);

    size_t len = strlen(input);
    while (len > 0) {
        unsigned char ch = (unsigned char)input[len - 1];
        if (ch == ' ' || ch == '\t' || ch == '\n' || ch == '\r' || ch == '\v' || ch == '\f') {
            len--;
        } else {
            break;
        }
    }

    char *trimmed = (char *)malloc(len + 1);
    if (!trimmed) {
        fprintf(stderr, "Error: allocation failed\n");
        return 2;
    }
    memcpy(trimmed, input, len);
    trimmed[len] = '\0';

    printf("%s\n", trimmed);

    free(trimmed);
    return 0;
}

File: java/gpt/p124_nre.java
public final class p124_nre {
    private static boolean isWhitespace(char c) {
        return c == ' ' || c == '\t' || c == '\n' || c == '\r' || c == '\f' || c == '\u000B';
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Error: expected exactly 1 argument");
            System.exit(1);
        }

        String input = args[0];
        System.out.println(input);

        int end = input.length();
        while (end > 0 && isWhitespace(input.charAt(end - 1))) {
            end--;
        }
        String trimmed = input.substring(0, end);

        System.out.println(trimmed);

        System.exit(0);
    }
}